wrappers/wasm/demo_reader.html

```diff
@@ -20,49 +60,81 @@
 
 		var buffer = zxing._malloc(fileData.length);
 		zxing.HEAPU8.set(fileData, buffer);
-		var result = zxing.readBarcodeFromImage(buffer, fileData.length, true, format);
+		var readerResults = zxing.readBarcodesFromImage(buffer, fileData.length, true, format, 0xff);
 		zxing._free(buffer);
 
-		showImage(document.getElementById("drop_zone"), fileData, file.type, result.position);
-		showScanResult(result);
+		showScanResults(readerResults);
+		showImageWithResults(document.getElementById("drop_zone"), fileData, file.type, readerResults.results);
 	}
 	reader.readAsArrayBuffer(file);
 }
 
-function showImage(container, fileData, fileType, position) {
+function showImageWithResults(container, fileData, fileType, results) {
 	fileType = fileType || "image/jpeg";
 	container.innerHTML = "";
 	var img = document.createElement("img");
 	img.addEventListener('load', function() {
+		const canvas = document.createElement("canvas");
 		container.style.width = img.width + 'px';
 		container.style.height = img.height + 'px';
-		const canvas = document.createElement("canvas");
 		canvas.width = img.width;
 		canvas.height = img.height;
 		const ctx = canvas.getContext("2d");
 		ctx.drawImage(img, 0, 0);
-		ctx.beginPath();
-		ctx.moveTo(position.topLeft.x, position.topLeft.y);
-		ctx.lineTo(position.topRight.x, position.topRight.y);
-		ctx.lineTo(position.bottomRight.x, position.bottomRight.y);
-		ctx.lineTo(position.bottomLeft.x, position.bottomLeft.y);
-		ctx.closePath();
-		ctx.strokeStyle = "red";
-		ctx.lineWidth = 5;
-		ctx.stroke();
+
+		// Example to use JavaScript Array created from an Iterator
+		const resultArray = Array.from(new ReaderResultVectorIterator(results));
```

Comment:
Creating an array with a copy of the data here, only to then iterate over it in an indexed for loop, does not make much sense to me. Please either use the .size(), .get() API or the `for( ... on ...)` syntax.